api/resolvers/serial.js

```diff
@@ -64,38 +81,20 @@ export default async function serialize (models, ...calls) {
     maxTimeout: 100,
     retries: 10
   })
-}
-
-export async function serializeInvoicable (query, { models, lnd, hash, hmac, me, enforceFee }) {
-  if (!me && !hash) {
-    throw new Error('you must be logged in or pay')
-  }
-
-  let trx = Array.isArray(query) ? query : [query]
-
-  let invoice
-  if (hash) {
-    invoice = await checkInvoice(models, hash, hmac, enforceFee)
-    trx = [
-      models.$executeRaw`SELECT confirm_invoice(${hash}, ${invoice.msatsReceived})`,
-      ...trx
-    ]
-  }
-
-  let results = await serialize(models, ...trx)
 
   if (hash) {
-    if (invoice?.isHeld) { await settleHodlInvoice({ secret: invoice.preimage, lnd }) }
+    if (invoice?.isHeld) {
+      await settleHodlInvoice({ secret: invoice.preimage, lnd })
+    }
     // remove first element since that is the confirmed invoice
-    [, ...results] = results
+    results = results.slice(1)
   }
 
-  // if there is only one result, return it directly, else the array
-  results = results.flat(2)
```

Comment:
Did not include this line in the new merged code since I couldn't find why we needed it. It was added in 10203f565. I funded my wallet, founded a territory, changed billing from 100k/month to 1m/year, zapped around using NWC, everything worked fine.

I understand what it's supposed to do (unflat arrays) but seems unnecessary or even like a source for bugs. It might have been related to our weird array input/output parsing which no longer exists. If you put in an array, you now get back an array and vice versa. Before, the output did not directly depend on the input but on the output length which was weird.

Reply:
Did everything except create a simple comment, tunnel vision was strong with this one haha